dafni/main_dafni.py

```diff
@@ -0,0 +1,210 @@
+
+
+class ValidationError(Exception):
+    """
+    Custom class to capture validation errors in this script
+    """
+    pass
+
+
+def get_args(test_args=None) -> argparse.Namespace:
+    """
+    Function to parse arguments from the user using the CLI
+    :param test_args: None
+    :returns:
+            - argparse.Namespace - A Namsespace consisting of the arguments to this script
+    """
+    parser = argparse.ArgumentParser(description="A script for running the causal testing famework on DAFNI.")
+
+    parser.add_argument(
+        "--data_path", required=True,
+        help="Path to the input runtime data (.csv)", nargs="+")
+
+    parser.add_argument('--tests_path', required=True,
+                        help='Path to the input configuration file containing the causal tests (.json)')
+
+    parser.add_argument('--variables_path', required=True,
+                        help='Path to the input configuration file containing the predefined variables (.json)')
+
+    parser.add_argument("--dag_path", required=True,
+                        help="Path to the input file containing a valid DAG (.dot). "
+                             "Note: this must be supplied if the --tests argument isn't provided.")
+
+    parser.add_argument('--output_path', required=False, help='Path to the output directory.')
+
+    parser.add_argument(
+        "-f",
+        default=False,
+        help="(Optional) Failure flag to step the framework from running if a test has failed.")
+
+    parser.add_argument(
+        "-w",
+        default=False,
+        help="(Optional) Specify to overwrite any existing output files. "
+             "This can lead to the loss of existing outputs if not "
+             "careful")
+
+    args = parser.parse_args(test_args)
+
+    # Convert these to Path objects for main()
+
+    args.variables_path = Path(args.variables_path)
+
+    args.tests_path = Path(args.tests_path)
+
+    if args.dag_path is not None:
+
+        args.dag_path = Path(args.dag_path)
+
+    if args.output_path is None:
+
+        args.output_path = "./outputs/results/"+"_".join([os.path.splitext(os.path.basename(x))[0]
+                                                          for x in args.data_path]) + "_results.json"
+
+        Path(args.output_path).parent.mkdir(exist_ok=True, parents=True)
+
+    else:
+
+        args.output_path = Path(args.output_path)
+
+        args.output_path.parent.mkdir(exist_ok=True, parents=True)
+
+    return args
+
+
+def read_variables(variables_path: Path) -> dict:
+    """
+    Function to read the variables.json file specified by the user
+    :param variables_path: A Path object of the user-specified file path
+    :returns:
+            - dict - A valid dictionary consisting of the causal tests
+    """
+    if not variables_path.exists() or variables_path.is_dir():
+
+        raise ValidationError(f"Cannot find a valid settings file at {variables_path.absolute()}.")
```

Comment:
I'd argue this could be the built in `FileNotFoundError ` rather than a custom exception.